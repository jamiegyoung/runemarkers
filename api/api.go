package api

import (
	"os"

	"github.com/jamiegyoung/runemarkers-go/entities"
	"github.com/jamiegyoung/runemarkers-go/logger"
	"github.com/jamiegyoung/runemarkers-go/pageio"
)

var log = logger.New("api")

func Generate(found_entities []*entities.Entity) {
	GenerateButtons(found_entities)
}

type ButtonPage struct {
	pageio.Page
	TilesString string
}

func (p ButtonPage) Data() map[string]interface{} {
 return map[string]interface{}{
  "Page": &p.Page,
  "TilesString": p.TilesString,
  }
}


func GenerateButtons(found_entities []*entities.Entity) {
	log("Generating buttons api")

	button_string, err := pageio.ReadPageString("./api/button.tmpl")
	if err != nil {
		panic(err)
	}

	// create the api folder if it doesn't exist
	if _, err := os.Stat("public/api"); os.IsNotExist(err) {
		mkdirerr := os.Mkdir("public/api", 0755)
		if mkdirerr != nil {
			panic(mkdirerr)
		}
	}

	for _, entity := range found_entities {
		out_file := pageio.CreateOutFile(
			"public/api",
			"api/button_"+entity.ApiUri+".html",
		)
		defer out_file.Close()

		page_data := ButtonPage{
      TilesString: entity.TilesString,
			Page: pageio.Page{
				ShowInfoButton: false,
			},
		}

		pageio.RenderPage(
			"button_"+entity.ApiUri,
			button_string,
			out_file,
			page_data,
		)
	}
}
