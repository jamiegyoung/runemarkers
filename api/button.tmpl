<button
  class="entity-card-copy animate"
  onclick="copy(this);
    this.classList.remove('animate');
    window.setTimeout(() => this.classList.add('animate'), 1);"
  data-tiles="{{ . }}"
>Copy</button>
