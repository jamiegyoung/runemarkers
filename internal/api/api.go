package api

import (
	"os"
	"path/filepath"
	"strings"

	"github.com/jamiegyoung/runemarkers-go/internal/entities"
	"github.com/jamiegyoung/runemarkers-go/internal/logger"
	"github.com/jamiegyoung/runemarkers-go/internal/pageio"
	"github.com/jamiegyoung/runemarkers-go/internal/pages"
)

var log = logger.New("api")

func Generate(ents []*entities.Entity) error {
	api_templates := []string{
		"templates/api/button.tmpl",
		"templates/api/tile_data_display.tmpl",
		"templates/api/tile_data_display_pretty.tmpl",
		"templates/api/tile_data_display_truncated.tmpl",
		"templates/api/tile_data_display_pretty_truncated.tmpl",
	}

	var err error = nil
	for _, tmpl := range api_templates {
		log("Generating entity specific api for " + tmpl)
		err = GenerateEntitySpecific(ents, tmpl)
		if err != nil {
			log("Error generating entity specific api: " + err.Error())
			break
		}
	}
	return err
}

func GenerateEntitySpecific(ents []*entities.Entity, tmpl_path string) error {
	api_page, err := pageio.ReadPageString(tmpl_path)
	if err != nil {
		return err
	}

	path_with_ext := filepath.Base(tmpl_path)
	api_name := strings.TrimSuffix(path_with_ext, filepath.Ext(path_with_ext))
	log("Generating " + api_name + " api")

	// create the api folder if it doesn't exist
	if _, err := os.Stat("public/api"); os.IsNotExist(err) {
		err = os.Mkdir("public/api", 0755)
		if err != nil {
			return err
		}
	}

	for _, entity := range ents {
		output, err := pageio.CreateOutFile(
			"public/api",
			"api/"+api_name+"_"+entity.ApiUri+".html",
		)
		if err != nil {
			return err
		}
		defer output.Close()

		data := pages.NewPage(
			map[string]interface{}{
				"TilesString":       entity.TilesString,
				"TilesStringPretty": entity.TilesStringPretty,
				"SafeApiUri":        entity.SafeApiUri,
			},
		)

		err = pageio.RenderHtml(
			api_name+"_"+entity.ApiUri,
			api_page,
			output,
			data,
		)
		if err != nil {
			return err
		}
	}
	return nil
}
