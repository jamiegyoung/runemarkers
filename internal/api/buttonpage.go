package api

import "github.com/jamiegyoung/runemarkers-go/internal/pageio"

type ButtonPage struct {
	Page        pageio.Page
	TilesString string
}

func (p ButtonPage) Data() map[string]interface{} {
	return map[string]interface{}{
		"Page":        &p.Page,
		"TilesString": p.TilesString,
	}
}
