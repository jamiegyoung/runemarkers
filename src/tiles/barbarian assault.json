{
  "name": "Barbarian Assault",
  "thumbnail": "https://oldschool.runescape.wiki/images/Barbarian_Assault_logo.jpg",
  "wiki": "https://oldschool.runescape.wiki/w/Barbarian_Assault",
  "tags": ["minigame","defender","healer"],
  "recommendedGuideVideoId": "Q_auOReO_2Y",
  "source": {
    "name": "Barbarian Assault for Beginners (OSRS)",
    "link": "https://www.youtube.com/watch?v=Q_auOReO_2Y"
  },
  "tiles": [
    {
      "regionId": 7509,
      "regionX": 29,
      "regionY": 46,
      "z": 0,
      "color": "#FF8E0F0F"
    },
    {
      "regionId": 7509,
      "regionX": 28,
      "regionY": 15,
      "z": 0,
      "color": "#FF746464"
    },
    {
      "regionId": 7509,
      "regionX": 27,
      "regionY": 16,
      "z": 0,
      "color": "#FF746464"
    },
    {
      "regionId": 7509,
      "regionX": 28,
      "regionY": 16,
      "z": 0,
      "color": "#FF746464",
      "label": "FILL VIAL"
    },
    {
      "regionId": 7509,
      "regionX": 45,
      "regionY": 29,
      "z": 0,
      "color": "#FF8E0F0F",
      "label": "2 Bad"
    },
    {
      "regionId": 7509,
      "regionX": 45,
      "regionY": 35,
      "z": 0,
      "color": "#FF0F8E2C",
      "label": "4 Good"
    },
    {
      "regionId": 7509,
      "regionX": 35,
      "regionY": 43,
      "z": 0,
      "color": "#FF0F8E2C",
      "label": "4 Good"
    },
    {
      "regionId": 7509,
      "regionX": 39,
      "regionY": 39,
      "z": 0,
      "color": "#FF8E0F0F",
      "label": "2 Bad"
    },
    {
      "regionId": 7508,
      "regionX": 44,
      "regionY": 34,
      "z": 0,
      "color": "#FF0F8E1E"
    },
    {
      "regionId": 7508,
      "regionX": 38,
      "regionY": 42,
      "z": 0,
      "color": "#FF0F8E1E"
    },
    {
      "regionId": 7508,
      "regionX": 30,
      "regionY": 46,
      "z": 0,
      "color": "#FF746464"
    }
  ]
}
