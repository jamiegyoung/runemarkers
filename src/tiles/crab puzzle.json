{
  "name": "Crab Puzzle",
  "thumbnail": "https://oldschool.runescape.wiki/images/thumb/Jewelled_Crab.png/120px-Jewelled_Crab.png?5c8ba",
  "wiki": "https://oldschool.runescape.wiki/w/Jewelled_Crab",
  "tags": ["cox", "chambers of xeric", "raids"],
  "recommendedGuideVideoId": "YplIdCg1qJs",
  "source": {
    "name": "WeDoRaids Discord & @caerulescens on GitHub",
    "link": "https://discord.gg/wdr",
    "modified": "Missing tiles added by @caerulescens"
  },
  "tiles": [
    {
      "regionId": 13395,
      "regionX": 7,
      "regionY": 46,
      "z": 2,
      "color": "#FF000000",
      "label":"Anchor"
    },
    {
      "regionId":13395,
      "regionX":8,
      "regionY":48,
      "z":2,
      "color":"#FF000000",
      "label":"Anchor"
    },
    {
      "regionId":13395,
      "regionX":14,
      "regionY":47,
      "z":2,
      "color":"#FFFFFFFF"
    },
    {
      "regionId":13395,
      "regionX":11,
      "regionY":56,
      "z":2,
      "color":"#FFFF0000"
    },
    {
      "regionId":13395,
      "regionX":11,
      "regionY":46,
      "z":2,
      "color":"#FF1FA347"
    },
    {
      "regionId":13395,
      "regionX":10,
      "regionY":46,
      "z":2,
      "color":"#FFFF0000"
    },
    {
      "regionId":13395,
      "regionX":10,
      "regionY":47,
      "z":2,
      "color":"#FF001DFF"
    },
    {
      "regionId":13139,
      "regionX":14,
      "regionY":40,
      "z":2,
      "color":"#FF001DFF"
    },
    {
      "regionId":13139,
      "regionX":17,
      "regionY":40,
      "z":2,
      "color":"#FF1FA347"
    },
    {
      "regionId":13139,
      "regionX":18,
      "regionY":40,
      "z":2,
      "color":"#FFFFFFFF"
    },
    {
      "regionId":13139,
      "regionX":22,
      "regionY":40,
      "z":2,
      "color":"#FFFF0000"
    },
    {
      "regionId":13139,
      "regionX":25,
      "regionY":41,
      "z":2,
      "color":"#FF000000",
      "label":"Anchor"
    },
    {
      "regionId":13139,
      "regionX":24,
      "regionY":39,
      "z":2,
      "color":"#FF000000",
      "label":"Anchor"
    },
    {
      "regionId":13139,
      "regionX":54,
      "regionY":42,
      "z":2,
      "color":"#FFFF0000"
    },
    {
      "regionId":13139,
      "regionX":40,
      "regionY":43,
      "z":2,
      "color":"#FFFF0000"
    },
    {

      "regionId":13139,
      "regionX":54,
      "regionY":43,
      "z":2,
      "color":"#FFFFFFFF"
    },
    {

      "regionId":13139,
      "regionX":51,
      "regionY":47,
      "z":2,
      "color":"#FF0000FF"
    },
    {

      "regionId":13139,
      "regionX":52,
      "regionY":49,
      "z":2,
      "color":"#FF0000FF"
    },
    {

      "regionId":13139,
      "regionX":54,
      "regionY":46,
      "z":2,
      "color":"#FF0000FF"
    },
    {

      "regionId":13139,
      "regionX":54,
      "regionY":47,
      "z":2,
      "color":"#FF00FF00"
    },
    {

      "regionId":13139,
      "regionX":41,
      "regionY":48,
      "z":2,
      "color":"#FF00FF00"
    }
  ]
}
