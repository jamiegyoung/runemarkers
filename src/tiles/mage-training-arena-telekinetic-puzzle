{
  "name": "MTA Telekinetic Puzzle Tiles",
  "thumbnail": "https://oldschool.runescape.wiki/images/thumb/Telekinetic_Guardian.png/156px-Telekinetic_Guardian.png",
  "wiki": "https://oldschool.runescape.wiki/w/Mage_Training_Arena/Telekinetic_Theatre",
  "tags": ["skilling", "magic", "mta", "mage training arena"],
  "recommendedGuideVideoId": "",
  "source": {
    "name": "",
    "link": ""
  },
  "tiles": [
    {
      "regionId": 13463,
      "regionX": 10,
      "regionY": 10,
      "z": 2,
      "color": "#FF94F0C8",
      "label": "2,6"
    },
    {
      "regionId": 13463,
      "regionX": 10,
      "regionY": 19,
      "z": 2,
      "color": "#FF94F0C8",
      "label": "4,8"
    },
    {
      "regionId": 13463,
      "regionX": 11,
      "regionY": 9,
      "z": 2,
      "color": "#FF94F0C8",
      "label": "3,7"
    },
    {
      "regionId": 13463,
      "regionX": 20,
      "regionY": 20,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "7"
    },
    {
      "regionId": 13463,
      "regionX": 20,
      "regionY": 11,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "1"
    },
    {
      "regionId": 13463,
      "regionX": 10,
      "regionY": 40,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "7"
    },
    {
      "regionId": 13463,
      "regionX": 20,
      "regionY": 50,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "4,10"
    },
    {
      "regionId": 13463,
      "regionX": 57,
      "regionY": 23,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "7"
    },
    {
      "regionId": 13463,
      "regionX": 47,
      "regionY": 13,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "4"
    },
    {
      "regionId": 13463,
      "regionX": 13,
      "regionY": 43,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "1,5,7"
    },
    {
      "regionId": 13463,
      "regionX": 12,
      "regionY": 53,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "2"
    },
    {
      "regionId": 13463,
      "regionX": 13,
      "regionY": 54,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "3"
    },
    {
      "regionId": 13463,
      "regionX": 12,
      "regionY": 44,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "4,6,8"
    },
    {
      "regionId": 13463,
      "regionX": 51,
      "regionY": 18,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "8"
    },
    {
      "regionId": 13463,
      "regionX": 48,
      "regionY": 56,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "2,6"
    },
    {
      "regionId": 13463,
      "regionX": 37,
      "regionY": 56,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "4,8"
    },
    {
      "regionId": 13463,
      "regionX": 11,
      "regionY": 20,
      "z": 2,
      "color": "#FF94F0C8",
      "label": "1,5,9"
    },
    {
      "regionId": 13463,
      "regionX": 55,
      "regionY": 59,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "1,5"
    },
    {
      "regionId": 13463,
      "regionX": 56,
      "regionY": 49,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "2,6"
    },
    {
      "regionId": 13463,
      "regionX": 55,
      "regionY": 48,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "3,7"
    },
    {
      "regionId": 13463,
      "regionX": 9,
      "regionY": 20,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "5"
    },
    {
      "regionId": 13463,
      "regionX": 10,
      "regionY": 21,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "4"
    },
    {
      "regionId": 13463,
      "regionX": 57,
      "regionY": 12,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "1"
    },
    {
      "regionId": 13463,
      "regionX": 58,
      "regionY": 13,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "6"
    },
    {
      "regionId": 13463,
      "regionX": 58,
      "regionY": 22,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "2"
    },
    {
      "regionId": 13463,
      "regionX": 38,
      "regionY": 57,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "7"
    },
    {
      "regionId": 13463,
      "regionX": 43,
      "regionY": 57,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "3,5"
    },
    {
      "regionId": 13463,
      "regionX": 47,
      "regionY": 57,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "1"
    },
    {
      "regionId": 13463,
      "regionX": 49,
      "regionY": 23,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "3"
    },
    {
      "regionId": 13463,
      "regionX": 52,
      "regionY": 12,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "5"
    },
    {
      "regionId": 13463,
      "regionX": 51,
      "regionY": 9,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "4"
    },
    {
      "regionId": 13463,
      "regionX": 50,
      "regionY": 8,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "3"
    },
    {
      "regionId": 13463,
      "regionX": 40,
      "regionY": 9,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "6"
    },
    {
      "regionId": 13463,
      "regionX": 40,
      "regionY": 18,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "10"
    },
    {
      "regionId": 13463,
      "regionX": 19,
      "regionY": 15,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "3"
    },
    {
      "regionId": 13463,
      "regionX": 28,
      "regionY": 16,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "4,8"
    },
    {
      "regionId": 13463,
      "regionX": 18,
      "regionY": 26,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "1"
    },
    {
      "regionId": 13463,
      "regionX": 16,
      "regionY": 51,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "3"
    },
    {
      "regionId": 13463,
      "regionX": 10,
      "regionY": 51,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "1"
    },
    {
      "regionId": 13463,
      "regionX": 9,
      "regionY": 49,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "6"
    },
    {
      "regionId": 13463,
      "regionX": 9,
      "regionY": 50,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "2"
    },
    {
      "regionId": 13463,
      "regionX": 9,
      "regionY": 45,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "8"
    },
    {
      "regionId": 13463,
      "regionX": 11,
      "regionY": 51,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "9"
    },
    {
      "regionId": 13463,
      "regionX": 42,
      "regionY": 19,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "7,9"
    },
    {
      "regionId": 13463,
      "regionX": 50,
      "regionY": 19,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "1"
    },
    {
      "regionId": 13463,
      "regionX": 25,
      "regionY": 45,
      "z": 2,
      "color": "#FF94F0C8",
      "label": "4"
    },
    {
      "regionId": 13463,
      "regionX": 24,
      "regionY": 44,
      "z": 2,
      "color": "#FF94F0C8",
      "label": "5"
    },
    {
      "regionId": 13463,
      "regionX": 25,
      "regionY": 49,
      "z": 2,
      "color": "#FF94F0C8",
      "label": "6"
    },
    {
      "regionId": 13463,
      "regionX": 23,
      "regionY": 55,
      "z": 2,
      "color": "#FF94F0C8",
      "label": "7"
    },
    {
      "regionId": 13463,
      "regionX": 14,
      "regionY": 54,
      "z": 2,
      "color": "#FF94F0C8",
      "label": "8"
    },
    {
      "regionId": 13463,
      "regionX": 15,
      "regionY": 55,
      "z": 2,
      "color": "#FF94F0C8",
      "label": "9"
    },
    {
      "regionId": 13463,
      "regionX": 14,
      "regionY": 45,
      "z": 2,
      "color": "#FF94F0C8",
      "label": "2"
    },
    {
      "regionId": 13463,
      "regionX": 13,
      "regionY": 10,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "2"
    },
    {
      "regionId": 13463,
      "regionX": 9,
      "regionY": 14,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "3"
    },
    {
      "regionId": 13463,
      "regionX": 16,
      "regionY": 21,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "6"
    },
    {
      "regionId": 13463,
      "regionX": 56,
      "regionY": 58,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "4"
    },
    {
      "regionId": 13463,
      "regionX": 17,
      "regionY": 44,
      "z": 2,
      "color": "#FF94F0C8",
      "label": "3"
    },
    {
      "regionId": 13463,
      "regionX": 15,
      "regionY": 44,
      "z": 2,
      "color": "#FF94F0C8",
      "label": "1"
    },
    {
      "regionId": 13463,
      "regionX": 26,
      "regionY": 26,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "5"
    },
    {
      "regionId": 13463,
      "regionX": 17,
      "regionY": 19,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "2"
    },
    {
      "regionId": 13463,
      "regionX": 51,
      "regionY": 12,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "2"
    },
    {
      "regionId": 13463,
      "regionX": 44,
      "regionY": 8,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "5"
    },
    {
      "regionId": 13463,
      "regionX": 13,
      "regionY": 51,
      "z": 0,
      "color": "#FF94F0C8",
      "label": "5"
    },
    {
      "regionId": 13463,
      "regionX": 17,
      "regionY": 23,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "6"
    },
    {
      "regionId": 13463,
      "regionX": 21,
      "regionY": 15,
      "z": 1,
      "color": "#FF94F0C8",
      "label": "7"
    }
  ]
}
