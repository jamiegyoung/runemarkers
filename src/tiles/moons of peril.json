{
  "name": "Moons of Peril",
  "thumbnail": "https://oldschool.runescape.wiki/images/thumb/Blood_Moon.png/120px-Blood_Moon.png",
  "wiki": "https://oldschool.runescape.wiki/w/Moons_of_Peril",
  "tags": [
    "boss"
  ],
  "tiles": [
    {
      "regionId": 5782,
      "regionX": 59,
      "regionY": 32,
      "z": 0,
      "color": "#FFEA8B33",
      "label": "STAB"
    },
    {
      "regionId": 5782,
      "regionX": 32,
      "regionY": 59,
      "z": 0,
      "color": "#FFA0E0F8",
      "label": "CRUSH"
    },
    {
      "regionId": 5782,
      "regionX": 4,
      "regionY": 32,
      "z": 0,
      "color": "#FFDB2867",
      "label": "SLASH"
    },
    {
      "regionId": 5783,
      "regionX": 31,
      "regionY": 13,
      "z": 0,
      "color": "#FFA0E0F8",
      "label": "STORM"
    },
    {
      "regionId": 5783,
      "regionX": 33,
      "regionY": 17,
      "z": 0,
      "color": "#FFA0E0F8",
      "label": "SAFE"
    },
    {
      "regionId": 5783,
      "regionX": 34,
      "regionY": 19,
      "z": 0,
      "color": "#FFA0E0F8",
      "label": "ICE"
    },
    {
      "regionId": 5783,
      "regionX": 45,
      "regionY": 18,
      "z": 0,
      "color": "#FFA0E0F8",
      "label": "SAFE"
    },
    {
      "regionId": 5783,
      "regionX": 45,
      "regionY": 14,
      "z": 0,
      "color": "#FFA0E0F8",
      "label": "SAFE"
    },
    {
      "regionId": 5783,
      "regionX": 19,
      "regionY": 18,
      "z": 0,
      "color": "#FFA0E0F8",
      "label": "SAFE"
    },
    {
      "regionId": 5783,
      "regionX": 19,
      "regionY": 14,
      "z": 0,
      "color": "#FFA0E0F8",
      "label": "SAFE"
    },
    {
      "regionId": 6038,
      "regionX": 18,
      "regionY": 35,
      "z": 0,
      "color": "#FFEA8B33",
      "label": "CLONES"
    },
    {
      "regionId": 6038,
      "regionX": 15,
      "regionY": 29,
      "z": 0,
      "color": "#FFEA8B33",
      "label": "ORB"
    },
    {
      "regionId": 5526,
      "regionX": 50,
      "regionY": 35,
      "z": 0,
      "color": "#FFDB2867",
      "label": "BLOOD"
    },
    {
      "regionId": 5526,
      "regionX": 46,
      "regionY": 29,
      "z": 0,
      "color": "#FFDB2867",
      "label": "JAGUAR"
    }
  ]
}