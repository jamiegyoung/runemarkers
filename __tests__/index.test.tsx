import { getTileData } from '@/api/tiles';
import '@testing-library/jest-dom';
import { act, fireEvent, render } from '@testing-library/react';
import Home, { getStaticProps } from '@/pages/index';

describe(`Home`, () => {
  it(`should search correctly`, () => {
    const data = getTileData();
    const { container, getByPlaceholderText } = render(
      <Home tileData={data} />,
    );
    const input = getByPlaceholderText(`search tiles`);
    act(() => {
      fireEvent.change(input, { target: { value: `abyssal sire` } });
    });
    expect(container).toMatchSnapshot();
  });

  it(`should generate static props`, () => {
    const tileData = getTileData();
    const { props } = getStaticProps();
    expect(props.tileData).toEqual(tileData);
  });
});
