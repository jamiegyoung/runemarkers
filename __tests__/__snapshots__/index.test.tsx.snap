// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should search correctly 1`] = `
<div>
  <div>
    <main
      class="main"
    >
      <div
        class="navbar"
      >
        <a
          class="link titleLink"
          href="/"
          rel=""
          target="_self"
        >
          <header
            class="title"
          >
            R
            <span>
              une
            </span>
            <span
              class="titleAlt"
            >
              M
              <span>
                arkers
              </span>
            </span>
            <span>
              .net
            </span>
          </header>
        </a>
      </div>
      <div
        class="container"
      >
        <p
          style="padding: 0px; color: rgb(199, 199, 199); text-align: center; min-width: 280px; margin-bottom: 15px; max-width: 600px; width: 80%;"
        >
          Make sure the "Ground Markers" plugin is enabled on RuneLite with "Remember color per tile" enabled. Copy the tile markers to your clipboard, right-click the world map button in-game and select "import".
        </p>
        <div
          class="listOperationsContainer grid"
        >
          <div
            style="display: flex; align-items: center; position: absolute; left: 50%; width: 100%; transform: translateX(-50%); justify-content: center;"
          >
            <input
              class="input "
              placeholder="search tiles"
              type="text"
              value="abyssal sire"
            />
            <button
              class="clearButton "
            >
              <svg
                height="24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6.4 19.45 4.55 17.6l5.6-5.6-5.6-5.6L6.4 4.55l5.6 5.6 5.6-5.6 1.85 1.85-5.6 5.6 5.6 5.6-1.85 1.85-5.6-5.6Z"
                />
              </svg>
            </button>
          </div>
          <button
            aria-label="Change layout"
            class="button"
            title="Change layout"
            type="button"
          >
            <div
              class="innerButton"
            >
              <div
                class="buttonSection grid"
              />
              <div
                class="buttonSection grid"
              />
              <div
                class="buttonSection grid"
              />
              <div
                class="buttonSection grid"
              />
              <div
                class="buttonSection grid"
              />
              <div
                class="buttonSection grid"
              />
            </div>
          </button>
        </div>
        <div
          class="innerGridContainer"
        >
          <div
            class="card"
          >
            <a
              class="imageContainer"
              href="/abyssal%20sire"
            >
              <img
                alt="Abyssal Sire image"
                class="image"
                data-nimg="1"
                decoding="async"
                height="120"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Foldschool.runescape.wiki%2Fimages%2Fthumb%2FAbyssal_Sire.png%2F120px-Abyssal_Sire.png&w=256&q=75"
                srcset="/_next/image?url=https%3A%2F%2Foldschool.runescape.wiki%2Fimages%2Fthumb%2FAbyssal_Sire.png%2F120px-Abyssal_Sire.png&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Foldschool.runescape.wiki%2Fimages%2Fthumb%2FAbyssal_Sire.png%2F120px-Abyssal_Sire.png&w=256&q=75 2x"
                style="color: transparent; opacity: 0;"
                width="120"
              />
            </a>
            <div
              class="cardData"
            >
              <div
                class="entityInfo"
              >
                <a
                  class="name"
                  href="/abyssal%20sire"
                >
                  Abyssal Sire
                </a>
              </div>
              <div
                class="tileInteraction"
              >
                <p
                  class="tileCount"
                >
                  24
                   tile markers
                </p>
                <button
                  class="button button"
                >
                  Copy
                  <div
                    class="animBar "
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <footer
        class="footer"
      >
        <div
          class="footerItem"
        >
          <p>
            Made by
            <a
              class="link "
              href="https://discord.com/users/167850724976361472"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="GitHub"
                data-nimg="1"
                decoding="async"
                height="17"
                loading="lazy"
                src="/_next/image?url=%2Fdiscord.png&w=48&q=75"
                srcset="/_next/image?url=%2Fdiscord.png&w=32&q=75 1x, /_next/image?url=%2Fdiscord.png&w=48&q=75 2x"
                style="color: transparent;"
                width="20"
              />
              <span
                class="linkText"
              >
                jamgyo
              </span>
            </a>
          </p>
        </div>
        <div
          class="footerItem"
        >
          <a
            class="link "
            href="https://github.com/jamiegyoung/runemarkers"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="GitHub"
              data-nimg="1"
              decoding="async"
              height="20"
              loading="lazy"
              src="/_next/image?url=%2Fgithub.png&w=48&q=75"
              srcset="/_next/image?url=%2Fgithub.png&w=32&q=75 1x, /_next/image?url=%2Fgithub.png&w=48&q=75 2x"
              style="color: transparent;"
              width="20"
            />
            <span
              class="linkText"
            >
              contribute code / tiles
            </span>
          </a>
        </div>
      </footer>
    </main>
  </div>
</div>
`;
